packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -198,39 +210,64 @@ export class Outbox {
 			return;
 		}
 
+		// Reference and/or Client sequence number will be advancing while processing this batch,
+		// so we can't use this check to detect wrongdoing. But we will still log via telemetry.
+		// This is rare, and the reentrancy will be handled during Flush.
+		const expectedDueToReentrancy = this.isContextReentrant();
+
+		const error = getLongStack(() =>
```

Comment:
If we do not throw and do not log, then we are paying the cost of grabbing stack for nothing - i believe it's not cheap.
Is it Ok?
Should this be a functor that is invoked only when it's needed, maybe?